File: app/src/main/java/com/awscherb/cardkeeper/ui/cards/CardsContract.java
package com.awscherb.cardkeeper.ui.cards;

import com.awscherb.cardkeeper.data.model.ScannedCode;
import com.awscherb.cardkeeper.ui.base.BasePresenter;
import com.awscherb.cardkeeper.ui.base.BaseView;

import java.util.List;

public interface CardsContract {

    interface View extends BaseView {

        void showCards(List<ScannedCode> codes);

        void onCardAdded(ScannedCode code);

        void onCardDeleted();

    }

    interface Presenter extends BasePresenter<CardsContract.View> {

        void loadCards();

        void addNewCard(ScannedCode code);

        void deleteCard(ScannedCode code);

    }


}

File: app/src/main/java/com/awscherb/cardkeeper/ui/cards/CardsContract.kt
package com.awscherb.cardkeeper.ui.cards

import com.awscherb.cardkeeper.data.model.ScannedCode
import com.awscherb.cardkeeper.ui.base.BasePresenter
import com.awscherb.cardkeeper.ui.base.BaseView

interface CardsContract {

    interface View : BaseView {

        fun showCards(codes: List<ScannedCode>)

        fun onCardAdded(code: ScannedCode)

        fun onCardDeleted()
    }

    interface Presenter : BasePresenter<CardsContract.View> {

        fun loadCards()

        fun addNewCard(code: ScannedCode)

        fun deleteCard(code: ScannedCode)
    }
}
